File: app/src/test/java/crosslanguage/ScalaServiceTest.java
package crosslanguage;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;
import services.scala.ScalaServiceApp;

public class ScalaServiceTest {
    @Test
    void testScalaService() {
        String greeting = ScalaServiceApp.greet();
        assertEquals("Hello from Scala Service!", greeting);
    }
}

File: ScalaService/bin/test/com/service/scala/ScalaServiceTest.scala
package com.service.scala

import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions.assertEquals
import com.service.scala.ScalaServiceApp

class ScalaServiceTest:
  @Test
  def testScalaService(): Unit =
    val greeting = ScalaServiceApp.greet()
    assertEquals("Hello from Scala Service!", greeting)